src/sched/sched_yield.c

```diff
@@ -1,18 +1,7 @@
 #include <sched.h>
-#include <enclave/lthread.h>
-#include "enclave/lthread_int.h"
 #include "syscall.h"
 
 int sched_yield()
 {
-        /* 
-         * lthread_yield does not add the current thread to the end of scheduler
-         * queue. We have to enqueue this thread after entering the scheduler
-         * because otherwise there is a chance of a race condition -- Another
-         * scheduler might pick up this lthread while the current scheduler is
-         * still executing it.
-         */
-        struct lthread *lt = lthread_self();
-        _lthread_yield_cb(lt, (void *)__scheduler_enqueue, lt);
-	return 0;
+	return syscall(SYS_sched_yield);
```

Comment:
I think this will work, but please can you add a unit test for yield that runs with one ethread and ensures that two pthreads can make progress?

Reply:
Added test in sgx-lkl at `basic/tests/pthread_yield`